server/src/main/java/io/druid/server/http/DatasourcesResource.java

```diff
@@ -551,8 +551,10 @@ public int compare(DruidDataSource druidDataSource, DruidDataSource druidDataSou
       }
 
       // segment stats
-      totalSegmentCount += druidDataSource.getSegments().size();
-      totalSegmentSize += dataSourceSegmentSize;
+      if (dataSourceSegmentSize >= totalSegmentSize) {
```

Comment:
@noddi yeah, looking through this code I wrote again, getSimpleDatasource iterates through all the servers and pulls out info about a specific datasource for each server. The information about segments in each tier includes replication, and the total segments includes replication as well. The dataSourceSegmentSize is tracking the total segment size on a per server basis and totalSegmentSize is tracking total segment size in the cluster. I don't fully understand the logic of the if statement.